client/client.go

```diff
@@ -25,568 +25,76 @@
-	height uint64,
-	opts ...grpc.CallOption,
-) (*flow.BlockHeader, error) {
-	req := &access.GetBlockHeaderByHeightRequest{
-		Height: height,
-	}
-
-	res, err := c.rpcClient.GetBlockHeaderByHeight(ctx, req, opts...)
-	if err != nil {
-		return nil, newRPCError(err)
-	}
-
-	return getBlockHeaderResult(res)
-}
-
-func getBlockHeaderResult(res *access.BlockHeaderResponse) (*flow.BlockHeader, error) {
-	header, err := convert.MessageToBlockHeader(res.GetBlock())
-	if err != nil {
-		return nil, newMessageToEntityError(entityBlockHeader, err)
-	}
-
-	return &header, nil
-}
-
-// GetLatestBlock gets the full payload of the latest sealed or unsealed block.
-func (c *Client) GetLatestBlock(
-	ctx context.Context,
-	isSealed bool,
-	opts ...grpc.CallOption,
-) (*flow.Block, error) {
-	req := &access.GetLatestBlockRequest{
-		IsSealed: isSealed,
-	}
-
-	res, err := c.rpcClient.GetLatestBlock(ctx, req, opts...)
-	if err != nil {
-		return nil, newRPCError(err)
-	}
-
-	return getBlockResult(res)
-}
-
-// GetBlockByID gets a full block by ID.
-func (c *Client) GetBlockByID(
-	ctx context.Context,
-	blockID flow.Identifier,
-	opts ...grpc.CallOption,
-) (*flow.Block, error) {
-	req := &access.GetBlockByIDRequest{
-		Id: blockID.Bytes(),
-	}
-
-	res, err := c.rpcClient.GetBlockByID(ctx, req, opts...)
-	if err != nil {
-		return nil, newRPCError(err)
-	}
-
-	return getBlockResult(res)
-}
-
-// GetBlockByHeight gets a full block by height.
-func (c *Client) GetBlockByHeight(
-	ctx context.Context,
-	height uint64,
-	opts ...grpc.CallOption,
-) (*flow.Block, error) {
-	req := &access.GetBlockByHeightRequest{
-		Height: height,
-	}
-
-	res, err := c.rpcClient.GetBlockByHeight(ctx, req, opts...)
-	if err != nil {
-		return nil, newRPCError(err)
-	}
-
-	return getBlockResult(res)
-}
-
-func getBlockResult(res *access.BlockResponse) (*flow.Block, error) {
-	block, err := convert.MessageToBlock(res.GetBlock())
-	if err != nil {
-		return nil, newMessageToEntityError(entityBlock, err)
-	}
-
-	return &block, nil
-}
-
-// GetCollection gets a collection by ID.
-func (c *Client) GetCollection(
-	ctx context.Context,
-	colID flow.Identifier,
-	opts ...grpc.CallOption,
-) (*flow.Collection, error) {
-	req := &access.GetCollectionByIDRequest{
-		Id: colID.Bytes(),
-	}
-
-	res, err := c.rpcClient.GetCollectionByID(ctx, req, opts...)
-	if err != nil {
-		return nil, newRPCError(err)
-	}
-
-	result, err := convert.MessageToCollection(res.GetCollection())
-	if err != nil {
-		return nil, newMessageToEntityError(entityCollection, err)
-	}
-
-	return &result, nil
-}
-
-// SendTransaction submits a transaction to the network.
-func (c *Client) SendTransaction(
-	ctx context.Context,
-	tx flow.Transaction,
-	opts ...grpc.CallOption,
-) error {
-	txMsg, err := convert.TransactionToMessage(tx)
-	if err != nil {
-		return newEntityToMessageError(entityTransaction, err)
-	}
-
-	req := &access.SendTransactionRequest{
-		Transaction: txMsg,
-	}
+type Client interface {
```

Comment:
Can we keep the `Close` function on this interface? I think it's actually pretty important for the gRPC implementation.

It can just be a no-op on `http.Client`.